src/boot/efi/boot.c

```diff
@@ -1985,6 +1985,52 @@ static const void *devicetree_orig;
 static EFI_PHYSICAL_ADDRESS devicetree_addr;
 static UINTN devicetree_pages;
 
+#ifdef __riscv
+#define DEVICETREE_OVERALLOC 64
+
+static EFI_STATUS devicetree_fixup(void *fdt, UINTN size) {
+        const fdt32_t *value;
+        int chosen;
+        int len;
+        uint32_t boot_hartid;
+
+        /* get /chosen/boot-hartid from original fdt */
+        chosen = fdt_subnode_offset_namelen(devicetree_orig, 0, "chosen", 6);
+        if (chosen < 0)
+                return EFI_INVALID_PARAMETER;
+
+        value = fdt_getprop_namelen(devicetree_orig, chosen, "boot-hartid", 11, &len);
+        if (value == NULL || len != sizeof(*value))
```

Comment:
we typically would write `!value` rather than `value == NULL`. see 

https://systemd.io/CODING_STYLE/#formatting

Reply:
Gotcha